app/lib/draggable.jsx

```diff
@@ -0,0 +1,102 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import classnames from 'classnames';
+
+function Draggable(props) {
+  let _previousEventCoords = {};
+  let moveEvent;
+  let endEvent;
+
+  function _rememberCoords(e) {
+    _previousEventCoords = {
+      x: e.pageX,
+      y: e.pageY
+    };
+  }
+
+  function handleStart(e) {
+    e.preventDefault();
+
+    switch (e.type) {
+      case 'mousedown':
+        [moveEvent, endEvent] = ['mousemove', 'mouseup'];
+        break;
+      case 'touchstart':
+        [moveEvent, endEvent] = ['touchmove', 'touchend'];
+        break;
+      default:
+        [moveEvent, endEvent] = ['pointermove', 'pointerup'];
+    }
+
+    e = (e.touches && e.touches[0]) ? e.touches[0] : e;
```

Comment:
Reassigning `e` here, to get the event (x,y) coords, is confusing later on. I keep thinking that the callback is called with the original event, but it isn't really.